src/Server.cpp

```diff
@@ -402,263 +689,146 @@ bool Server::checkForCompleteRequest(int clientSock)
- *
- * Adds a virtual server to the list of virtual servers and registers it with epoll.
- *
- * @param serverSock The socket of the virtual server to register.
- *
- * @return true if the virtual server was successfully registered, false otherwise.
- */
-bool Server::registerVirtualServer(const Socket& serverSock)
+void connectionReceiveBody(Server& server, int clientFd, Connection& connection)
 {
-	struct epoll_event event = {};
-	event.events = EPOLLIN;
-	event.data.fd = serverSock.fd;
+	LOG_DEBUG << "ReceiveBody for: " << connection.m_clientSocket;
 
-	if (!addEvent(m_epfd, serverSock.fd, &event)) {
-		close(serverSock.fd);
-		LOG_ERROR << "Failed to add event for " << serverSock;
-		return false;
-	}
-
-	m_virtualServers[serverSock.fd] = serverSock;
-
-	LOG_INFO << "New virtual server: " << serverSock;
-	return true;
+	(void)server;
+	(void)clientFd;
 }
 
 /**
- * @brief Register a connection.
- *
- * Registers a client fd with epoll.
- * If it fails to add the event, it logs an error and closes the client socket.
- * If the connection is successfully registered, adds the connection to map m_connections.
- * The []-operator creates a new entry in the map if the key does not exist or overwrites an existing one.
- * The same way (re-)initializes the map m_connectionBuffers[newFD] to an empty string.
+ * @brief Builds the response for a client connection.
  *
- * @param serverSock The socket of the server that the connection is associated with.
- * @param clientSock The socket of the client connection to register.
+ * Builds the response for a client connection by calling Server::buildResponse().
+ * The response is stored in the connection buffer.
+ * The connection status is set to SendResponse and the response is sent to the client by calling
+ * connectionSendResponse().
  *
- * @return true if the connection was successfully registered, false otherwise.
+ * @param server The server object which handles the connection.
+ * @param clientFd The file descriptor of the client.
+ * @param connection The connection object to build the response for.
  */
-bool Server::registerConnection(const Socket& serverSock, const Socket& clientSock)
+void connectionBuildResponse(Server& server, int clientFd, Connection& connection)
 {
-	struct epoll_event event = {};
-	event.events = EPOLLIN;
-	event.data.fd = clientSock.fd;
-
-	if (!addEvent(m_epfd, clientSock.fd, &event)) {
-		close(clientSock.fd);
-		LOG_ERROR << "Failed to add event for " << clientSock;
-		return false;
-	}
-
-	m_connections[clientSock.fd] = Connection(serverSock, clientSock);
-	m_connectionBuffers[clientSock.fd] = "";
+	LOG_DEBUG << "BuildResponse for: " << connection.m_clientSocket;
 
-	LOG_INFO << "New Connection: " << clientSock << " for server: " << serverSock;
-	return true;
+	server.buildResponse(connection.m_request);
+	connection.m_buffer = server.getResponse();
+	connection.m_status = Connection::SendResponse;
+	connectionSendResponse(server, clientFd, connection);
```

Comment:
Yes: If we only send half the message, the connection status stays at Connection::SendResponse. The next time epoll gives us this fd we can try to send the rest.

Reply:
Wouldn't it be cleaner to just call the connectionSendResponse() function in the switch case instead of once there and once here?

Reply:
The "loop" in this case is epoll notifying us that the client is _still_ ready to receive a response. 

We need to make one stop in BuildResponse(), but we can then try to send immediatly. I think in most cases this would lead to a successful send, setting the connection status to Close.

However if we go back to epoll we would need to be notified two times of the same file descriptor just to be able to send a single response. If we don't have many clients it makes no difference. But if we serve many clients, this seems like unneccesary overhead.

Reply:
Would then add a comment in the documentation to make this intention clear.

Reply:
Looks good to me, the purpose of each function is clear now. Short function name is sufficient.

Reply:
Added comment and changed to Connection::ReceiveHeader and connectionReceiveHeader()